third-party/crowdsec/src/main/java/org/apache/james/model/CrowdsecHttpClient.java

```diff
@@ -0,0 +1,99 @@
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+
+import javax.inject.Inject;
+
+import org.apache.james.util.AuditTrail;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.datatype.jdk8.Jdk8Module;
+
+import reactor.core.publisher.Mono;
+import reactor.netty.http.client.HttpClient;
+
+public class CrowdsecHttpClient {
+    private static final Logger LOGGER = LoggerFactory.getLogger(CrowdsecHttpClient.class);
+    private static final String GET_DECISION = "/decisions";
+
+    public static final int OK = 200;
+    public static final int UNAUTHENTICATED = 403;
+    public static final int NOT_FOUND = 404;
+
+    private final HttpClient httpClient;
+    private final ObjectMapper mapper;
+
+    @Inject
+    public CrowdsecHttpClient(CrowdsecClientConfiguration configuration) {
+        this.httpClient = buildReactorNettyHttpClient(configuration);
+        this.mapper = new ObjectMapper().registerModule(new Jdk8Module());
+    }
+
+    public Mono<List<CrowdsecDecision>> getCrowdsecDecisions() {
+        return httpClient.get()
+            .uri(GET_DECISION)
+            .responseSingle((response, body) -> {
+                switch (response.status().code()) {
+                    case OK:
+                        return body.asString().map(this::parseCrowdsecDecisions);
+                    case UNAUTHENTICATED:
+                        AuditTrail.entry()
+                            .log("Invalid api-key bouncer");
+                        return Mono.empty();
+                    case NOT_FOUND:
+                        AuditTrail.entry()
+                            .log("Crowdsec Url not found");
+                        return Mono.empty();
+                    default:
+                        return Mono.error(new RuntimeException("Request failed with status code " + response.status().code()));
+                }
+            });
+    }
+
+    private List<CrowdsecDecision> parseCrowdsecDecisions(String json) {
+        if (Objects.equals(json, "null")) {
+            return new ArrayList<>();
+        } else {
+            try {
+                return mapper.readValue(json, new TypeReference<>() {});
+            } catch (JsonProcessingException e) {
+                throw new RuntimeException(e);
+            }
+        }
+    }
+
+    private HttpClient buildReactorNettyHttpClient(CrowdsecClientConfiguration configuration) {
+        return HttpClient.create()
+            .responseTimeout(DEFAULT_TIMEOUT)
+            .baseUrl(configuration.getUrl().toString())
+            .headers(headers -> headers.add("X-Api-Key", configuration.getApiKey()))
+            .headers(headers -> headers.add("accept", "application/json"));
```

Comment:
Use a constant for Accept? (From a well known HTTP library?)

Reply:
I already researched and there is no built-in constant for Accept

Reply:
Didn't search well enough :)

From James code base for example: https://github.com/apache/james-project/blob/master/server/testing/src/main/java/org/apache/james/jmap/HttpJmapAuthentication.java#L61

Reply:
![image](https://github.com/apache/james-project/assets/55171818/2ea79f70-3752-4710-b69d-4c534e275174)